libraries/AP_AHRS/AP_AHRS_NavEKF.cpp

```diff
@@ -96,6 +96,18 @@ void AP_AHRS_NavEKF::update(bool skip_ins_update)
         update_EKF3();
         update_EKF2();
     }
+
+    // update home location for GPS baro correction for DCM
+    Location EKF_home_loc;
+    if (_ekf_type == 2 && _ekf2_started && (EKF2.getOriginHgtMode() & (1<<2)) && EKF2.healthy()) {
+        EKF2.getOriginLLH(-1, EKF_home_loc);
```

Comment:
@EShamaev Can you please explain what this PR fixes. Were you worried about a jump in height if the AHRS falls back to DCM? (only relevant for plane). I don't like it for the reason we are now introducing a dependancy on the EKF internal calculation in the fallback mode.